src/oop/object_interactions/medium/q4/QuestionText.md

```diff
@@ -0,0 +1,53 @@
+# oop-object_interactions-medium-q4
+
+- Java question
+
+## Question Text
+
+You are working as a vet assistant in an animal hospital, and sometimes you encounter an animal that requires a blood
+transfusion in order to save its life. Suppose you were given the following test class. Make a class called **Snail**
+with attributes **bloodType** and **volume** that are initialized by the constructor. This class has a method called
+**transfusionFrom()** that checks if the two animals have the same blood type and whether the donor has enough blood
+before starting the transfusion. If both conditions pass, then 1L of blood is transferred from the donor to the
+recipient.
+
+## Snail Class
+
+The `Snail` class currently has the following items:
+
+- Attributes:
+    - A `char` called **bloodType**
+    - A `double` called **volume**
+    - None of these attributes should be accessible from other classes.
+
+- Constructors:
+    - A constructor that take a `char` and a `double` to initialize the **bloodType** and **volume** attributes.
+    - It is accessible from other classes.
+
+- Methods:
+    - Getter ans Setter methods for **bloodType** and **volume**, and  **toString** method that prints the object
+      information.
+    - These methods are accessible from other classes.
+    -
+
+You must add the following items to the class:
+
+- Methods:
+    - A `transfusionFrom` method that take another **Snail** object to do a blood tansfer.
+    - It should check if the blood types are a match and if not print `Blood type incompatible`.
```

Comment:
should probably make a note here that snail blood types are not like humans, and to check if they match exactly or something, since human blood types are a bit more complicated for who can transfer blood to another person and we want to clear up any confusion